File: JS/BOJ-B3-10988-Check_Palindrome.js
"use strict";

const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "../input.txt";
let input = fs
  .readFileSync(filePath)
  .toString()
  .trim()
  .map((v) => v.trimEnd());

let output = [];
const solution = (input) => {
  let result = 1;
  const reInput = input.split('').reverse();
  for (let i = 0; i < input.length; ++i) {
    if (input[i] != reInput[i]) {
      result = 0;
      break;
    }
  }
  output.push(result);
};

solution(input);
fs.writeFileSync("/dev/stdout", output.join(""));
// fs.writeFileSync("../output.txt", output.join(""), (err) => console.error);

File: CPP/BOJ-B3-10988-Check_Palindrome.cpp
#include <bits/stdc++.h>
using namespace std;

// string str;
// int f = 1;

// int main()
// {
//   ios_base::sync_with_stdio(false);
//   cin.tie(NULL);
//   cout.tie(NULL);
//   freopen("input.txt", "r", stdin);
//   freopen("output.txt", "w", stdout);

//   cin >> str;

//   for (int i = 0; i < str.length(); ++i)
//   {
//     if (str[i] == str[str.length() - i - 1])
//       continue;
//     else
//     {
//       f = 0;
//       break;
//     }
//   }
//   cout << f;

//   return 0;
// }

string s, temp;

int main()
{
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);
  cout.tie(NULL);
  freopen("input.txt", "r", stdin);
  freopen("output.txt", "w", stdout);

  cin >> s;
  temp = s;
  reverse(temp.begin(), temp.end());
  if (temp == s)
    cout << 1 << "\n";
  else
    cout << 0 << "\n";

  return 0;
}